common/lib/common-definitions/src/logger.ts

```diff
@@ -59,6 +59,47 @@ export interface ITelemetryPerformanceEvent extends ITelemetryGenericEvent {
     duration?: number; // Duration of event (optional)
 }
 
+/**
+ * Broad classifications to be applied to individual properties as they're prepared to be logged to telemetry.
+ * Please do not modify existing entries for backwards compatibility.
+ */
+ export enum TelemetryDataTag {
+    /** Data containing terms from code packages that may have been dynamically loaded */
+    PackageData = "PackageData",
+    /** Personal data of a variety of classifications that pertains to the user */
+    UserData = "UserData",
+}
+
+/**
+ * A property to be logged to telemetry containing both the value and the tag
+ */
+export interface ITaggedTelemetryPropertyType {
+    value: TelemetryEventPropertyType,
+    tag: TelemetryDataTag
+}
+
+/**
+ * Property bag containing a mix of value literals and wrapped values along with a tag
+ */
+export interface ITaggableTelemetryProperties {
+    [name: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
+}
+
+/**
+ * Type guard to identify if a particular value (loosely) appears to be a tagged telemetry property
+ */
+export const isTaggedTelemetryPropertyValue = (x: any): x is ITaggedTelemetryPropertyType => {
```

Comment:
@vladsud I've wondered if it's better to have the type guard in the definitions package (so it is updated in lock-step with the interface if it changes) v. in the utils package (so we don't introduce any JS build output in the definitions package).  I'm leaning towards the utils package to keep the definitions package "pure", and there's not really harm in having it in a separate package since the dependency between utils/definitions is explicit at build time.

So @ksbrar I think my vote is to move this and the other type guard below to common-utils (in a subsequent PR once the common-definitions update publishes)

Reply:
Either way is fine, but I guess it's a bit cleaner to have in utils as definitions packages are not supposed to have any code

Reply:
Removed, to be added to utils in next PR.